### ISSUE TITLE
Normalizing a Duration larger than one month has an unexpected amount of days

### ISSUE THREAD

**Issue by johannaljunggren** [moment/luxon#1514]
**Describe the bug**
I have created a duration that is 28 days long. After I normalize it it has the value 1 month. When I get the value of the normalized duration in days it returns 30 days.

**To Reproduce**
https://codesandbox.io/s/luxon-duration-fx9v3l

**Actual vs Expected behavior**
const duration = Duration.fromObject({
  years: 0,
  months: 0,
  weeks: 0,
  days: 28,
  hours: 0,
  minutes: 0,
  seconds: 0,
  milliseconds: 0
});

const normalizedDuration = duration.normalize();
Expected: normalizedDuration.as('days') == 28;
Actual: normalizedDuration.as('days') == 30;

**Desktop (please complete the following information):**

- OS: [Windows 10]
- Browser [Firefox 117.0.1]
- Luxon version [3.4.3]
- Your timezone [CEST/GMT+2]

**Comment by icambron** [comment]
Yeah, that's wrong. The underlying issue is that it normalized 28 days to 1 month, which isn't correct. Then when you converted it back to days, 1 month turned into 30 days, which by itself is correct.

What I suspect is happening is that because you have a `weeks` key, 28 days gets converted to 4 weeks, like this:

```js
Duration.fromObject({ weeks: 0, days: 28 }).normalize().toObject()   //=> { weeks: 4, days: 0 }
```

Then, because you have a `months` key too, it says "well, _and_ 4 weeks is 1 month":

```js
Duration.fromObject({ months: 0, weeks: 0, days: 28 }).normalize().toObject()  //=> { months: 1, weeks: 0, days: 0 }
```

If you leave out months, it becomes 4 weeks, 0 days, which will convert back to 28 days when you do `as("days")`. And if you leave out weeks, it won't be a full month, so it'll act normally:

```js
Duration.fromObject({ months: 0, days: 28 }).normalize().toObject()  //=> { months: 0, days: 28 }
```

So it's the combo of `months` and `weeks` as keys in the duration that's trigging this.

Fundamentally, it's the recursive rollup is problematic, because the `casual` conversion rules aren't self-consistent; it does this only because the normalization "passes through" weeks, and 4 weeks = 1 month is inconsistent with 30 days = 1 month. I think we need to improve this. @diesieben07 I know you were recently looking at duration normalization stuff. Do you have thoughts on this one? My off-the-cuff thought is that we shouldn't always be rolling up already-partially-rolled-up units, but I'm not quite sure how to formalize that.

**Comment by danganiev-tt** [comment]
It seems that this issue was introduced while fixing `normalize` for negative values after [version 3.4.0 changes](https://github.com/moment/luxon/blob/master/CHANGELOG.md#340-2023-08-08), because unit tests in the project I'm working on, which uses `Duration` are passing for Luxon version 3.3.0

**Comment by diesieben07** [comment]
If we prevent this conversion then "normalize" is no longer idempotent:

`Duration.fromObject{ months: 0, weeks: 0, days: 28 }).normalize()` should become `{ months: 0, weeks: 4, days: 0 }` according to the claim. But if you then normalize _again_ you'll get `{ months: 1, weeks: 0, days: 0 }`, because by the 2nd normalize call the information that the 4 weeks came from 28 days is lost. I don't think this is a desirable quality at all.

This is just a fundamental limitation of the inconsistent rules in the (default) `casualMatrix`. It says that 28 days is 4 weeks and also that 4 weeks are 1 month, but then 1 month is also 30 days. The two contradict.
